linode/manager
# refactor: [M3-7920, M3-7930] – Refactor VPC queries

## Description 📝
Refactor the VPC queries to use `createQueryKeys` from `query-key-factory`.

The main objective was to achieve parity with existing functionality, although some small enhancements at the margins were done as well.

## Changes  🔄
- Use `createQueryKeys` to generate VPC & subnet query keys
- Update mock region data to include VPC capability
- Add new `useAllVPCsQuery` and use it in `VPCPanel.tsx` & `useVPCConfigInterface.tsx`
- Remove manual uses of `vpcQueryKey` and `subnetQueryKey` throughout codebase, notably for various query invalidations
- Remove unused `useSubnetQuery`

## Target release date 🗓️
4/15/24

## How to test 🧪
### Verification steps
Essentially the entire VPC offering in Cloud needs to be tested to ensure nothing has broken with these changes.

- VPC landing page
  - query for paginated results should refresh upon VPC creation
- Assign/Unassign Linode flow via drawer accessed from Subnets table
  - should invalidate the paginated VPC, individual VPC, and VPC subnets queries
- Assigning Linode to VPC via Linode Config dialog
  - all VPC queries should be invalidated
- Linode Create flow
  - a newly-created linode assigned to a VPC should have the VPC section present in the Linode Detail header
- Linode Detail page
  - the VPC section in the Linode Detail header should be accurate, and the "IP Addresses" table on the "Network" tab should contain the expected VPC-related IPs
- Deleting a Linode
  - the 'all' VPCs, paginated VPCs, individual VPC, and VPC subnets queries should be invalidated upon Linode deletion

## As an Author I have considered 🤔
